flow/project.py

```diff
@@ -3313,18 +3308,26 @@ def select(operation):
                 # Change groups to available run _JobOperation(s)
                 with self._potentially_buffered():
                     operations = []
-                    for flow_group in flow_groups:
-                        for aggregate in self._get_aggregate_store(
-                            flow_group.name
-                        ).values():
-                            operations.extend(
-                                flow_group._create_run_job_operations(
-                                    self._entrypoint,
-                                    default_directives,
-                                    aggregate,
-                                    ignore_conditions,
+                    run_groups = set(self._gather_flow_groups(names))
+                    for (
+                        aggregate_store,
+                        aggregate_groups,
+                    ) in self._group_to_aggregator.inverse.items():
+                        matching_groups = set(aggregate_groups) & run_groups
+                        if not matching_groups:
+                            # Skip iteration over aggregate store if no groups match the
+                            # specified names
+                            continue
+                        for aggregate in aggregate_store.values():
```

Comment:
```suggestion
                        for aggregate in aggregate_store.values():
                            if aggregate not in aggregates:
                                continue
```

This is a really small optimization which makes a difference when only a few aggregates/jobs are provided by the user.

Given this, we can now remove the exact same check in the `select` method above.

Reply:
I will look at this in #416, I refactored this code and may have already made the change you're suggesting.

Reply:
I think the new approach in #416 is similar to what you were looking for here. Please comment on that PR in the method `_generate_selected_aggregate_groups` if not. Here's the logic:
```python
if selected_aggregates is not None:
    # Use selected aggregates in the aggregate store
    for aggregate in aggregate_progress_wrapper(selected_aggregates):
        aggregate_id = get_aggregate_id(aggregate)
        if aggregate_id in aggregate_store:
            for group in matching_groups:
                yield aggregate_id, aggregate, group
else:
    # Use all aggregates in the aggregate store
    for aggregate_id, aggregate in aggregate_progress_wrapper(
        aggregate_store.items()
    ):
        for group in matching_groups:
            yield aggregate_id, aggregate, group
```